src/Tools/dotnet-monitor/EndpointInfo/IServerEndpointTracker.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Extensions.Hosting;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Threading;
+using System;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal record class EndpointRemovedEventArgs(IEndpointInfo Endpoint, ServerEndpointState State);
+
+    internal interface IServerEndpointTracker
+        : IHostedService
```

Comment:
This (`IHostedService`) should not be the base of the `IServerEndpointTracker` interface. It is an implementation detail of the `ServerEndpointTracker` implementation. Please remove and make sure that the implementation is registered as both an `IHostedService` and an `IServerEndpointTracker`.

Reply:
Agreed, this PR is currently draft since I'm mostly looking for PR builds to pass before opening it up

Reply:
Sorry, missed that it was a draft, or maybe it changed while I was reviewing it.